featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -461,7 +461,32 @@ def test_abides_by_max_depth_param(es):
         features = dfs_obj.build_features()
         for f in features:
             # last feature is identity feature which doesn't count
-            assert (f.get_depth() <= i + 1)
+            assert (f.get_depth() <= i)
+
+
+def test_max_depth_single_table(transform_es):
+    assert len(transform_es.entity_dict) == 1
+    for i in [0, 1, 2]:
+        def make_dfs_obj(max_depth=i):
+            dfs_obj = DeepFeatureSynthesis(target_entity_id='first',
+                                           entityset=transform_es,
+                                           trans_primitives=[AddNumeric],
+                                           max_depth=max_depth)
+            return dfs_obj
+
+        if i == 2:
+            match = ("Only one entity in entityset, changing max_depth to 1 "
+                     "since deeper features cannot be created")
+            with pytest.warns(UserWarning, match=match):
+                dfs_obj = make_dfs_obj(i)
+        else:
+            dfs_obj = make_dfs_obj(i)
+
+        features = dfs_obj.build_features()
+        # check transform features made if max_depth>=1
+        assert any([f.get_depth() == min(i, 1) for f in features])
+        # check all features depth 1 or less
+        assert all([f.get_depth() <= min(i, 1) for f in features])
```

Comment:
Is `min` needed here? Can we update to `assert all([f.get_depth() <= 1 for f in features])` instead?